jdbc/src/main/java/org/apache/metamodel/jdbc/dialects/HiveQueryRewriter.java

```diff
@@ -35,6 +35,10 @@ public String rewriteColumnType(ColumnType columnType, Integer columnSize) {
         if (columnType == ColumnType.INTEGER) {
             return "INT";
         }
+        // Hive does not support VARCHAR without a width, nor VARCHAR(MAX).
+        if (columnType == ColumnType.STRING && columnSize == null) {
```

Comment:
No I wouldn't worry about it. There was a bug once about not rewriting STRING properly but I think that's just history.